zb-db/src/main/java/io/camunda/zeebe/db/ColumnFamily.java

```diff
@@ -119,6 +119,13 @@
    */
   void delete(KeyType key);
 
+  /**
+   * Deletes the key-value pair with the given key if it exists in the column family
+   *
+   * @throws IllegalStateException if the key does not exist
+   */
+  void deleteExisting(KeyType key);
+
```

Comment:
💭 As replacement to `put` a new `upsert` method was introduced, which allowed the deprecation of `put`. Which makes me wonder whether we can find any ways to entice devs to favor `deleteExisting` over `delete`.

❓Do you see any ways to nudge our devs towards deleteExisting?

Reply:
Good point, that's something I have thought about a bit. When I started on this branch I added `ensureDeleted` and `deleteExisting` which would have allowed me to mark `delete` as deprecated as well. Then I tried to use the new methods everywhere and noticed that it is fairly difficult to judge whether to use `ensureDeleted` or `deleteExisting`. At the same time, I thought that the value of having and using checked delete methods was lower than the value of checked insert methods. 
That's why I then removed `ensureDeleted` and just left `delete` as is, non-deprecated, but left `deleteExisting` in case it is useful somewhere.

Do you think we should have an `ensureDeleted` method?

Reply:
That's a really good point. I honestly don't know, but I lean towards adding an `ensureDeleted` simply to make us think about each individual case (even if that's hard), and for consistency with `put`, `insert` and `upsert`. But, I'm fine either way. 

Perhaps @saig0 or @npepinpe wants to way in with their opinion

Reply:
I've added a `deleteIfExists` method and deprecated `delete`. Thanks for the suggestion :+1: